### PR TITLE
emb6: deprecate package

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
I basically didn't work on `emb6` since 2016 and adapting to the newest version would mean some major overhaul. However, the development at their end seems to be stalled [since March 2018][emb6-develop] as well. All this speaks for deprecating this package.

[emb6-develop]: https://github.com/hso-esk/emb6/tree/develop

### Testing procedure
Read, compile `make doc`, it should show up in the deprecated list (`doc/doxygen/html/deprecated.html`)

### Issues/PRs references
None

### CODE REVIEW

File: pkg/emb6/doc.txt [hunk 1]
```diff
@@ -5,6 +5,10 @@
  * @brief    emb6 network stack
  * @see      https://github.com/hso-esk/emb6/blob/14e4a3cfff01644e078870e14e16a1fe60dcc895/doc/pdf/emb6.pdf
  *
+ * @deprecated  Based on very old version of `emb6` and the package is
+ *              basically unmaintained; will be removed after the 2020.06
```
Comment (kb2ma) [inline pkg/emb6/doc.txt]: Shouldn't this be 2020.07?

Comment (miri64) [reply pkg/emb6/doc.txt]: Yeah.

**Comment by aabadie** [comment]
> Am I missing something?

See #8664 emb6 documentation is not parsed by doxygen.

**Comment by miri64** [comment]
Mh... it was excluded in https://github.com/RIOT-OS/RIOT/pull/8665, but as far as I can see there is just one element that isn't documented. Should be a straight forward fix. Will do.

**Comment by miri64** [comment]
Fixed the documentation for `emb6` in general and addressed @kb2ma's comment.

**Comment by kb2ma** [comment]
Fix is good. The package builds now. I haven't used emb6, but I can accept that the package documentation is good. Even if not, it's a start and can be extended if emb6 is revived.

Please squash.

**Comment by miri64** [comment]
Squashed

**Pull request merged**